contributor/04-mechanics-of-contributing.md

```diff
@@ -0,0 +1,149 @@
+
+# Article
+
+Are you ready to start contributing to other teams projects/repos?
+Do you look forward to reducing your blockers not by management escalation but by collaboration? 
+This section gives practical advice and highlights gotchas to remember when making an InnerSource contribution. It enables you and the host team to have as pleasant an experience as possible, setting the foundation for more contributions and great collaboration.  
+
+This article is separated into the three steps you will likely experience
+
+- Soliciting your contribution opportunity and preparing to work on it
+- Actually creating the contribution
+- Polishing and wrapping the gift nicely and presenting it to the host team.
+
+If your contribution is larger, you'll possibly go through (some) these steps repeatedly as you iterate towards your common goal.
+It is very likely that as you do this everything will feel more and more naturally - maybe you'll even wonder why you were doing anything else before.
+
+## Preparing to work
+
+### Lead times
+
+One key difference is the turn-around time. 
+With every first time contribution you are coming to a new (host) team.
+As a result, you'll need to get to know their code-base and the technology used, but also their preferred development environment (think test framework, build system).
+Even in cases where this type of tooling is standardized each team will have developed some individual peculiarities.
+In addition to the technical side, you may be faced with differences in communication, think code-reviews.
+Even if you are coming back after a while the teams' ways and members might have changed.
+Take your time as you would to catch up with a friend you haven't seen in a while and whom you are visiting now.
+
+Give yourself enough lead time.
+Start early enough so that your work is available for you to leverage at the time you need it. 
+It's better to add more slack time initially - you'll get a feeling about the turn-around times once you work with the host team.
+Often, you will notice a reduction in turn-around time per host team after making a few successful contributions to that host team.
+This is effect can be observed with Open Source as well, you can read more about it [here](https://doi.org/10.1109/MS.2013.95). 
+
+[EDITOR NOTE: There was actually a paper on how trust is build up on remote/asynchronous collaboration. Maybe I'll find that again. 
+The basic point is that aside of reliability in collaboration, continued presence the effect of water cooler discussions and food/drink consumption is quite impressive...
+]  
+
+Especially in the case of larger distances between you and the host teams, e.g. multiple time zones difference as in San Francisco, USA (PDT) and Berlin, Germany (CEST), you will need to actively consider this and explicitly manage your and their expectations on reaction times.
```

Comment:
Added a paragraph on expectation management turned around in the way @MaineC suggested. What do you think?

```suggestion
### Expectation management
In your classic teams everyone had an idea of the expectable lead times.
Within an InnerSource context this might not be the case, either due to large time-zone differences (e.g. Seattle, USA with PDT vs Berlin, Germany with CEST) or you not being available full-time as with your original team even if they are in the same physical location as you are.
Thus, to prevent frustration on both sides, impatience and other non-trust-building effects you'll need to explicitly do expectations managemnet with regards to your expected reaction times.
One approach be to just quickly react with a "I'll look into it, I won't get to it in the next few days though" to a Trusted Committers feedback if you know that you'll only be able to come back to them in a few days.
Ideally you can provide them with a rough estimate when you likely will have time to take a look at their input.
Doing so builds trust by reliability even over non-physical contact, longer distance or otherwise asynchronous media.
Established trust will allow you overcome uncertainty bumps in the collaborative road ahead of you. 
```

Reply:
LGTM

Reply:
LGTM